### PR TITLE
2nd attempt at increasing difficulty

### PR DESCRIPTION
**Pull request by AwesomePatrol**
### Brief description of the changes:
Replacement of #159 and #157 by @Max01000 with the following improvements:
* voted curve is applied to the easiest difficulty only,
* mutagen effectiveness is updated in GUI only when needed.

### Tested Changes:
- [X] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

File: maps/biter_battles_v2/ai.lua [hunk 1]
```diff
@@ -438,6 +438,10 @@ Public.raise_evo = function()
 	if global.freeze_players then return end
 	if not global.training_mode and (#game.forces.north.connected_players == 0 or #game.forces.south.connected_players == 0) then return end
 	if game.ticks_played < 7200 then return end
+	if global.difficulty_vote_index == 1 then
+		local x = game.ticks_played/3600
```
Comment (cogito123) [inline maps/biter_battles_v2/ai.lua:442]: The divide here by 3600 is causing this calculation to yield the same value. It will be constantly 0.25 with some fractions affected on nth place.
https://www.wolframalpha.com/input/?i=plot%28%28%28x%2F3600%2F470%29%5E3.7%29%2B0.25%2C+0..720%29

Doing instead: ticks * 0.025 will give you:
```
curve value after 1 h : 0.252208
curve value after 2 h : 0.278691
curve value after 3 h : 0.378613
curve value after 4 h : 0.622870
curve value after 5 h : 1.101382
curve value after 6 h : 1.921456
curve value after 7 h : 3.206637
curve value after 8 h : 5.095830
curve value after 9 h : 7.742599
curve value after 10 h : 11.314587
curve value after 11 h : 15.993024
```

**Review by cogito123 — CHANGES_REQUESTED** [review]
Only one remark to curve function

File: maps/biter_battles_v2/ai.lua (hunk 1 above)
Comment (cogito123) [reply maps/biter_battles_v2/ai.lua:442]: Multiplying ticks by 0.015 will get it closer to what was presented from that excel sheet.
```
curve value after 0 h : 0.250000
curve value after 1 h : 0.250333
curve value after 2 h : 0.254334
curve value after 3 h : 0.269429
curve value after 4 h : 0.306327
curve value after 5 h : 0.378613
curve value after 6 h : 0.502496
curve value after 7 h : 0.696640
curve value after 8 h : 0.982028
curve value after 9 h : 1.381858
curve value after 10 h : 1.921456
curve value after 11 h : 2.628197
curve value after 12 h : 3.531447
curve value after 13 h : 4.662505
```

**Comment by cogito123** [comment]
Commit message in `Use new difficulty increase curve` needs some more work. Discord link is hard to follow outside of our little group. If it becomes invalid, developers will lose information about "why" it was changed.

File: maps/biter_battles_v2/ai.lua (hunk 1 above)
Comment (AwesomePatrol) [reply maps/biter_battles_v2/ai.lua:442]: Yes, if you plot it for the first 720 `ticks`, it will be `0.25` with some fractions. `x` is the length of the current match in minutes. The main objective was to get 100% evo around 7 hours mark, which the current curve definition does. If you want to change the curve to something that is different from the spreadsheet, please start a new vote.

Comment (cogito123) [reply maps/biter_battles_v2/ai.lua]: My bad, I assumed different units. It's OK.

**Comment by cogito123** [comment]
Tested on my machine, works great.

**Review by cogito123 — APPROVED** [review]
